packages/EasyErrorHandler/src/Processors/ErrorCodesGroupProcessor.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyErrorHandler\Processors;
+
+use EonX\EasyErrorHandler\DataTransferObjects\ErrorCodesDto;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesGroupProcessorInterface;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
+
+final class ErrorCodesGroupProcessor implements ErrorCodesGroupProcessorInterface
+{
+    public function __construct(
+        private int $categorySize,
+        private ErrorCodesProviderInterface $errorCodesProvider
+    ) {
+    }
+
+    public function process(): ErrorCodesDto
+    {
+        $groupedErrorCodes = [];
+
+        foreach ($this->errorCodesProvider->provide() as $errorCodeName => $errorCodeValue) {
+            $errorCodeCategory = $errorCodeValue - ($errorCodeValue % $this->categorySize);
+            $groupedErrorCodes[$errorCodeCategory] = $groupedErrorCodes[$errorCodeCategory] ?? [];
+            $groupedErrorCodes[$errorCodeCategory][$errorCodeName] = $errorCodeValue;
+        }
+
+        \ksort($groupedErrorCodes);
+
+        if (\count($groupedErrorCodes) === 0) {
+            return new ErrorCodesDto();
+        }
+
+        $nextGroupErrorCode = (int)\max(\array_keys($groupedErrorCodes)) + $this->categorySize;
+        $nextGroupedErrorCodes = [];
+
+        foreach ($groupedErrorCodes as $errorCodes) {
+            $nextGroupedErrorCodes[] = [
+                'categoryName' => $this->determineCategoryName(\array_keys($errorCodes)),
```

Comment:
What's the point of introducing a DTO if we still use some hard-coded array keys inside it? :wink: Let's add one more `ErrorCodeCategoryDto` class with `categoryName` and `nextErrorCodeToUse` properties and instantiate it here.

Reply:
Done.